### PR TITLE
MVP-16: Updated documentation

### PR DESCRIPTION
**Pull request by BradyMitch**
## 🎯 Summary

  
[MVP-16](https://citz-imb.atlassian.net/jira/browse/MVP-16)

- Updated flow diagram
- Updated README
- Updated Wiki (as temporary documentation before TechDocs is available on DevHub)

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,469 +8,54 @@
...
-2. Set up the package by following the steps at [Basic Setup Guide](#basic-setup-guide).
-3. For use with [@bcgov/citz-imb-sso-react].
-
-</details>
-
----
+# `Introduction`
 
-## Table of Contents
+This npm package offers a seamless integration solution for React applications requiring authentication through the B.C. government's Single Sign-On SSO [CSS] service. Designed with ease of use in mind, this package abstracts the complexity of handling SSO protocols, providing developers with simple, high-level APIs to facilitate secure user authentication and authorization within their applications. By leveraging this package, developers can quickly implement a robust authentication flow, ensuring that their React applications are securely integrated with the B.C. government's standardized identity management system, thereby enhancing user experience and ensuring compliance with government security standards.
```
Comment (BradyMitch) [inline README.md:13]:
```suggestion
This npm package offers a seamless integration solution for Express applications requiring authentication through the B.C. government's Single Sign-On SSO [CSS] service. Designed with ease of use in mind, this package abstracts the complexity of handling SSO protocols, providing developers with simple, high-level APIs to facilitate secure user authentication and authorization within their applications. By leveraging this package, developers can quickly implement a robust authentication flow, ensuring that their React applications are securely integrated with the B.C. government's standardized identity management system, thereby enhancing user experience and ensuring compliance with government security standards.
```

File: README.md [hunk 2]
```diff
@@ -8,469 +8,54 @@
...
-// Users must have EITHER 'Member' or 'Verified' role.
-if (hasRoles(user, ['Member', 'Verified'], { requireAllRoles: false })) // Do Something...
-```
+<br />
 
-User state can be accessed through `req.user`.
+#### `For Developers working on this package:`
 
-Example `req.user` object (Typescript Type is `SSOUser`):
+In our documentation, we emphasize the inclusion of detailed explanations on our code to significantly enhance onboarding efficiency, ensuring that new team members can quickly become productive by understanding the project's architecture and operational dynamics without extensive hand-holding.
```
Comment (TaylorFries) [inline README.md:35]:
```suggestion
In our documentation, we emphasize the inclusion of detailed explanations on our code to significantly enhance onboarding efficiency, ensuring that new team members can quickly become productive by understanding the project's architecture and operational dynamics. 
```

File: README.md [hunk 3]
```diff
@@ -8,469 +8,54 @@
...
-  "display_name": "Doe, John CITZ:EX",
-  "first_name": "John",
-  "last_name": "Doe",
-  "client_roles": ["Admin"],
-  "scope": "openid idir email profile azureidir",
-  "identity_provider": "idir",
-  "originalData": { /* ... (original user data from sso) */ }
-}
-```
+By making our project's infrastructure transparent and accessible, we encourage collaboration, enabling developers to easily navigate and contribute to different parts of the codebase without unnecessary barriers. Ultimately, this practice of enriching our documentation contributes to creating a self-documenting codebase, where the project's structure and operational commands are clearly defined and updated within the repository itself, making the project more maintainable and scalable over time.
```
Comment (TaylorFries) [inline README.md:37]:
```suggestion
We encourage collaboration by making our project's infrastructure transparent and accessible. Enabling developers to easily navigate and contribute to different parts of the codebase without barriers. Enriching our documentation contributes to a self-documenting codebase. The project's structure and commands are defined and updated within the repository itself. Ultimately, making the project more maintainable and scalable over time.
```

File: README.md [hunk 4]
```diff
@@ -8,469 +8,54 @@
...
-[CSS]: https://bcgov.github.io/sso-requests
-[SSO Keycloak Wiki - Identity Provider Attribute Mapping]: https://github.com/bcgov/sso-keycloak/wiki/Identity-Provider-Attribute-Mapping
+[example apps]: https://github.com/bcgov/keycloak-example-apps
+[CITZ IMB Common Code]: mailto:citz.codemvp@gov.bc.ca?subject=SSO%20Packages%20Support
+[Purpose and Benefits of This Package]: https://github.com/bcgov/citz-imb-sso-express/wiki/Purpose-and-Benefits-of-This-Package
+
+[Documentation]: https://github.com/bcgov/citz-imb-sso-express/wiki
+[Directory Structure]: https://github.com/bcgov/citz-imb-sso-express/wiki/b2-Directory-Structure
+[Scripts]: https://github.com/bcgov/citz-imb-sso-express/wiki/b3-Scripts
+[Publishing a New Release Version]: https://github.com/bcgov/citz-imb-sso-express/wiki/b4-Publishing-a-New-Release-Version
```
Comment (BradyMitch) [inline README.md:61]:
```suggestion
[Directory Structure]: https://github.com/bcgov/citz-imb-sso-express/wiki/11-Directory-Structure
[Scripts]: https://github.com/bcgov/citz-imb-sso-express/wiki/12-Scripts
[Publishing a New Release Version]: https://github.com/bcgov/citz-imb-sso-express/wiki/13-Publishing-a-New-Release-Version
```

File: README.md [hunk 5]
```diff
@@ -8,469 +8,54 @@
...
-2. Set up the package by following the steps at [Basic Setup Guide](#basic-setup-guide).
-3. For use with [@bcgov/citz-imb-sso-react].
-
-</details>
-
----
+# `Introduction`
 
-## Table of Contents
+This npm package offers a seamless integration solution for Express applications requiring authentication through the B.C. government's Single Sign-On SSO ([CSS]) service. Designed with ease of use in mind, this package abstracts the complexity of handling SSO protocols, providing developers with simple, high-level APIs to facilitate secure user authentication and authorization within their applications. By leveraging this package, developers can quickly implement a robust authentication flow, ensuring that their React applications are securely integrated with the B.C. government's standardized identity management system, thereby enhancing user experience and ensuring compliance with government security standards.
```
Comment (TaylorFries) [inline README.md:13]:
```suggestion
This npm package offers a high-level API on the B.C. government's Single Sign-On SSO ([CSS]) service for Express applications. High-level APIs abstract the complexity of handling SSO protocols for authentication and authorization. By using this package developers can create a secure Express application and meet B.C government security standards.
```

Comment (BradyMitch) [reply README.md:13]: How about this:

This npm package offers a simple integration solution for Express applications requiring authentication through the B.C. government's Single Sign-On SSO ([CSS]) service. It abstracts the complexity of handling SSO protocols manually. By using this package, developers can quickly implement authentication and authorization in their Express applications to meet B.C. government security standards.

Comment (TaylorFries) [reply README.md:13]: take the 'simple' out of the first line (and change a to an) and we have a deal :D

Comment (BradyMitch) [inline README.md:13]:
```suggestion
This npm package offers an integration solution for Express applications requiring authentication through the B.C. government's Single Sign-On SSO ([CSS]) service. It abstracts the complexity of handling SSO protocols manually. By using this package, developers can quickly implement authentication and authorization in their Express applications to meet B.C. government security standards.
```

**Review by TaylorFries — APPROVED** [review]
perfection

**Pull request merged**